pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -193,32 +193,29 @@ def on_run_end(self) -> List[List[STEP_OUTPUT]]:
         Raises:
             MisconfigurationException: ``train_epoch_end`` does not return ``None``
         """
-        if self.batch_progress.current.ready == 0:
```

Comment:
if I remove this check in the commit of PR #7871 then the test
`test_iterable_dataset_stop_iteration_at_epoch_beginning`
fails, however here it does not.

just an observation.